app/main.py

```diff
@@ -1,2 +1,53 @@
+from typing import Hashable
+
+
 class Dictionary:
-    pass
+    DEFAULT_CAPACITY = 8
+    LOAD_FACTOR = 2 / 3
+
+    def __init__(self,
+                 capacity: int = DEFAULT_CAPACITY,
+                 load_factor: float = LOAD_FACTOR) -> None:
+        self.capacity = capacity
+        self.load_factor = load_factor
+        self.length = 0
+        self.hash_table = [None] * capacity
+
+    def __len__(self) -> int:
+        return self.length
+
+    def __setitem__(self, key: Hashable, value: any) -> None:
+        index = hash(key) % self.capacity
+        while self.hash_table[index] is not None:
+            if self.hash_table[index][0] == key:
+                self.hash_table[index] = (key, value)
+                return
+            # handle collision
+            index = (index + 1) % self.capacity
+
+        self.hash_table[index] = (key, value)
+        self.length += 1
+
+        if self.length / self.capacity >= self.load_factor:
```

Comment:
The condition for resizing the hash table should be checked before inserting a new item to avoid exceeding the load factor. Consider moving this check to before line 28.